### ISSUE TITLE
Add local DIE handler to eval of json decode

### ISSUE THREAD

**Issue by Tanoti** [hirose31/AWS-CLIWrapper#15]
When using this module inside a larger piece of code that has a DIE handler installed, I found that should the json->decode_prefix() command fail inside the eval block for s3 commands (as expected I believe), the outer DIE handler caught it rather than allowing the eval block to work, The fix should be to add a local handler inside eval?

```
@@ -218,6 +218,7 @@ sub _execute {
         local $@;
         my($ret) = eval {
             # aws s3 returns null HTTP body, so failed to parse as JSON
+            $SIG{__DIE__} = sub {};
             $self->json->decode_prefix($json);
         };
```

Thanks for a very useful module!

**Comment by hirose31** [comment]
Does this PR https://github.com/hirose31/AWS-CLIWrapper/pull/16/files resolve your problem?

**Comment by Tanoti** [comment]
That's brilliant thanks! I was not confident enough with git to raise a PR myself.

**Comment by hirose31** [comment]
I've just released v1.14, it will be soon available.
